cmd/datamon/cmd/cli_fuse_test.go

```diff
@@ -3,166 +3,331 @@
 package cmd
 
 import (
+	"bufio"
+	"io"
 	"io/ioutil"
+	"math/rand"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"regexp"
+	"strconv"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
+	"github.com/oneconcern/datamon/pkg/dlogger"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
 )
 
-func testBundleMount(t *testing.T, testType string) {
+var (
+	buildBinaryonce sync.Once
+	testBinary      string
+)
+
+func testLogger(t testing.TB) *zap.Logger {
+	// a logger for the test sequence itself
+	return dlogger.MustGetLogger("debug").With(zap.String("test", t.Name()))
+}
+func testCommand(t testing.TB, target string, args ...string) (*exec.Cmd, io.ReadCloser) {
+	// NOTE: this starts a process, not a goroutine like for other tests
+	// That is primarily because we want to send a signal to the process
+	// in order to unmount.
+	cmd := exec.Command(target, args...)
+	cmd.Env = append(cmd.Env, "DATAMON_CONTEXT="+os.Getenv("DATAMON_CONTEXT")) // TODO(fred): not sure this is used
```

Comment:
is this environment variable forwarding what was preventing the tests from running in ci?

i fully appreciate the difficulty of the changes here to block based on stdio rather than a dumb timeout.

it's that i'm having difficulty separating the changes necessary to replace the dumb timeout with the blocking function created by `testMountReady` for those necessary to `Exec` the fuse command in ci.

Reply:
I am not sure about DATAMON_CONTEXT (since this is normally overridden with a flag in the tests to run next). I've to tidy up that a bit later on (we already have had a talk about propagating env variables).

 The main takeaway here is that we need to propagate the env to the forked process, esp. GOOGLE_APPLICATION_CREDENTIALS

Reply:
follow-up note: open discussion thread

Reply:
> The main takeaway here is that we need to propagate the env to the forked process

understood.

i think i realized this during the daemonization but forgot that it was also the issue with the fuse tests.. at time of writing i'd forgotten that the default wasn't to copy the current environment variables when calling `Exec()`.